src/main/webapp/app/overview/discussion-section/discussion-section.component.ts

```diff
@@ -90,28 +92,64 @@ export class DiscussionSectionComponent extends CourseDiscussionDirective implem
         this.postCreateEditModal?.modalRef?.close();
     }
 
+    /**
+     * on changing the sort direction via icon, the metis service is invoked to deliver the posts for the currently set context,
+     * sorted on the backend
+     */
+    onChangeSortDir(): void {
+        switch (this.currentSortDirection) {
+            case undefined: {
+                this.currentSortDirection = SortDirection.ASCENDING;
+                break;
+            }
+            case SortDirection.ASCENDING: {
+                this.currentSortDirection = SortDirection.DESCENDING;
+                break;
+            }
+            default: {
+                this.currentSortDirection = undefined;
+                break;
+            }
+        }
+        this.posts.sort(this.sectionSortFn);
+    }
+
     /**
      * sorts posts by following criteria
      * 1. criterion: displayPriority is PINNED -> pinned posts come first
      * 2. criterion: displayPriority is ARCHIVED  -> archived posts come last
      * -- in between pinned and archived posts --
-     * 3. criterion: vote-emoji count -> posts with more vote-emoji counts comes first
-     * 4. criterion: creationDate -> most recent comes at the end (chronologically from top to bottom)
+     * 3. criterion: creationDate - if activated by user through the sort arrow -> most recent comes at the end (chronologically from top to bottom)
+     * 4. criterion: if 3'rd criterion was not activated by the user, vote-emoji count -> posts with more vote-emoji counts comes first
+     * 5. criterion: if 3'rd criterion was not activated by the user, most recent posts comes at the end (chronologically from top to bottom)
```

Comment:
**5. criterion:** Shouldn't it be the 4th criterion that has not been activated in the JavaDoc? Of course, that also counts for the 3rd criterion implicitly, because a criterion with a higher number is never used if a criterion with a lower number is true.

Reply:
A criterion with a higher number may also be used if the evaluation of the previous criterion is even. So for example; let's say the user decided not to order by date, hence the 3rd criterion is not activated. In such context, posts are first ordered according to the number of upvotes and their creation date.

However, if the 3rd criterion was activated, then posts are ordered according to the creation date.

Reply:
Thank you for pointing it out. Yes, it seems confusing to me as well. I have changed the last criterion now to handle posts with the same amount of upvotes.